### ISSUE TITLE
Providers with pessimistic fog-core dependencies prevent bundling on edge

### ISSUE THREAD

**Issue by tokengeek** [fog/fog#4000]
Still getting failures on Travis with `gemfiles/Gemfile-edge` after fixing `fog-json` in #3999

```
Resolving dependencies...
Bundler could not find compatible versions for gem "fog-core":
  In Gemfile-edge:
    fog was resolved to 2.0.0, which depends on
      fog-core (~> 1.45)

    fog was resolved to 2.0.0, which depends on
      fog-json was resolved to 1.2.0, which depends on
        fog-core
```

So master fog/fog@610c179 currently has `fog-core ~> 1.45` and fog-core/master fog/fog-core@7865ef7 declares itself as v2.1 (which bundler does not show and would have made this obvious).

So `fog` master is not currently compatible with `fog-core` master.

The fix is complicated as a number of providers also depend on `fog-core ~> 1.45` and are included by `fog`

I just released `fog-brightbox v0.15.1` which releases the version dependencies there so we are no longer one of the offending providers.

The providers that are preventing this from resolving are thus:

- [x] fog-aliyun
- [x] fog-cloudatcost
- [x] fog-digitalocean
- [ ] fog-joyent
- [x] fog-ovirt
- [x] fog-powerdns

**Comment by plribeiro3000** [comment]
Most of those providers are not active. I will check and do it myself if needed.

**Comment by tokengeek** [comment]
@plribeiro3000 Thanks.

I just removed the version constraints from `fog-brightbox` altogether... https://github.com/fog/fog-brightbox/commit/fef450301b2e50ae4f67de7a4b357958fc51af7b

My thinking being make the versioning more of an application's concern but I'm not sure how that will scale out frankly.

**Comment by plribeiro3000** [comment]
Ops. Not supposed to close yet. 😄

**Comment by plribeiro3000** [comment]
@geemus `fog-cloudatcost` is fixed on github master but there is yet another provider i do not have access to release. Could you please add me up? Thanks

**Comment by plribeiro3000** [comment]
Okay. It took me some time and a lot of updates but now `fog-cloudatcost` is ready for a new release with a green (and working) build.

**Comment by geemus** [comment]
Thanks! Just pushed cloudatcost 0.4.0.

**Comment by plribeiro3000** [comment]
Now i'm struggling with `fog-digitalocean`. The code is so old and have so many discrepancies that its harsh to make the build work and without it i can say for sure i wont break it with the release of the dependency constraints. =s

**Comment by geemus** [comment]
@plribeiro3000 I can try to do it instead, if you want to see if the others are less painful?

**Comment by plribeiro3000** [comment]
sure. =)

**Comment by plribeiro3000** [comment]
@geemus Warning here. No one have access to [`fog-joyent`](https://rubygems.org/gems/fog-joyent) and it is declared as a dependency on `fog`.

Maybe we should just drop it from `fog` until we can work on releasing it?

**Comment by geemus** [comment]
Yeah, as we want to move toward including less anyway, probably not a bad plan. I also pinged on https://github.com/fog/fog-joyent/issues/1 (asking for access). So maybe it will also resolve in that way.

**Comment by plribeiro3000** [comment]
@geemus It seems i'm also lacking write access to `fog-ovirt`. Could you please add me in?

**Comment by geemus** [comment]
done

**Comment by plribeiro3000** [comment]
Awesome. Thanks!

**Comment by plribeiro3000** [comment]
Alright. We only have `joyent` and `digitalocean` to go now

**Comment by tas50** [comment]
Anything end users could do to help get those last 2 updated? We at Chef would love to be able to bundle in the latest fog-core with our development kit, but it includes joyent/digitalocean so we're riding the old release.

**Comment by plribeiro3000** [comment]
@tas50 You can send a PR to `fog-digitalocean` and we can merge and release but unfortunatelly `fog-joyent` is kind of dead right now and none of the `core` maintainers of `fog` have access to its gem on `rubygems`.

Only possible solution for this case is to drop it from `fog`. @geemus Any objections?

**Comment by tas50** [comment]
@plribeiro3000 Is this what you're looking for? https://github.com/fog/fog-digitalocean/pull/25

**Comment by plribeiro3000** [comment]
Ops. sorry about that. Reviewing now!

**Comment by plribeiro3000** [comment]
@tas50 Done. Some small comments to address otherwise good to go!

**Comment by plribeiro3000** [comment]
@tas50 Released `fog-openstack` version `0.4.0`

**Comment by geemus** [comment]
Thanks!
